### PR TITLE
[ServiceBus] remove network trace params from extra in pyamqp logging

### PR DESCRIPTION
**Pull request by swathipil**
related to #32190
following #32215 

TODO:
- [x] sync changes with EH pyamqp + README

### CODE REVIEW

**Comment by kashifkhan** [comment]
can we add the same changes in to EH as well please 🙂

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py [hunk 1]
```diff
@@ -723,7 +725,7 @@ async def _client_ready_async(self):
         if not self._link:
             self._link = self._session.create_receiver_link(
                 source_address=self.source,
-                link_credit=self._link_credit,
+                link_credit=0,  # link_credit=0 on flow frame sent before client is ready
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py:728]: calling out explicitly that the copy over is bringing the prefetch/link_credit = 0 fix to EH.

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py]: moved all non-logging changes to separate pr: #34407

**Comment by swathipil** [comment]
closing in favor of #34458

**Pull request closed without merging**